app/lib/BaseSearchController.php

```diff
@@ -238,9 +238,12 @@ public function Index($pa_options=null) {
 				$this->view->setVar('show_type_id', $vn_show_type_id);
 				$vo_result->filterResult('ca_objects.type_id', $vn_show_type_id);
 			}
-	
+
+			// Refresh result list in context to ensure it has the current "page" of results.
+			$this->opo_result_context->setResultList($vo_result->getPrimaryKeyValues());
+			$this->opo_result_context->setTotalCount($vo_result->numHits());
```

Comment:
Add 'total_count' to result context as current logic assumes all values are in the 'result_list' not a single page of results.